test/e2e_node/device_plugin_test.go

```diff
@@ -589,6 +589,57 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 				framework.Fail(fmt.Sprintf("stale device assignment after pod deletion while kubelet was down allocated=%v error=%v", allocated, err))
 			}
 		})
+
+		ginkgo.It("Can schedule a pod with a restartable init container [NodeAlphaFeature:SidecarContainers]", func(ctx context.Context) {
+			podRECMD := fmt.Sprintf("devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep %s", sleepIntervalWithRestart)
+			pod1 := e2epod.NewPodClient(f).CreateSync(ctx, makeBusyboxPodWithRestartableInitContainer(SampleDeviceResourceName, podRECMD))
```

Comment:
Thanks for adding e2e tests for this. Similar to my comment for the unit test, can we also add a non-restartable init container here as well to check if the resources were reclaimed back as expected?

Reply:
done.